### PR TITLE
Use vector transfer forwarding and dead store elimination in IREE

### PR DESCRIPTION
**Pull request by ThomasRaoux**

### CODE REVIEW

**Review by antiagainst — APPROVED** [review]
Awesome!

File: iree/compiler/Conversion/Common/Passes.h [hunk 1]
```diff
@@ -23,5 +23,8 @@ std::unique_ptr<OperationPass<ModuleOp>> createLegalizeNumWorkgroupsFnPass();
 /// each entry point function. The function is defined, but is populated later.
 std::unique_ptr<OperationPass<ModuleOp>> createDeclareNumWorkgroupsFnPass();
 
+/// Pass to optimize vector transfer_read and transfer_write.
+std::unique_ptr<FunctionPass> createVectorTransferOptimizationPass();
```
Comment (MaheshRavishankar) [inline iree/compiler/Conversion/Common/Passes.h:27]: Can we just call this VectorLoadStoreForwardingPass? The name is too generic :)

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]

**Review by MaheshRavishankar — APPROVED** [review]
Looks good! Ill leave the pass name upto you :)

**Comment by ThomasRaoux** [comment]
> Looks good! Ill leave the pass name upto you :)

Thanks Mahesh, I'll leave it like that for now to be consistent with upstream.

**Pull request merged**